x-pack/plugins/case/server/routes/api/cases/comments/get_comment.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { AllCommentsResponseRt, CommentResponseRt } from '../../../../../common/api';
+import { RouteDeps } from '../../types';
+import { flattenCommentSavedObject, flattenCommentSavedObjects, wrapError } from '../../utils';
+
+export function initGetCommentApi({ caseService, router }: RouteDeps) {
+  router.get(
+    {
+      path: '/api/cases/{case_id}/comments',
+      validate: {
+        params: schema.object({
+          case_id: schema.string(),
+        }),
+        query: schema.object({
+          id: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      try {
+        if (request.query.id != null && request.query.id.length > 0) {
```

Comment:
so if an id is provided, just return that one comment. otherwise return all as an array? should one endpoint really be returning two different types?

Reply:
yes it will return array of one